dsgrid/registry/project_registry_manager.py

```diff
@@ -642,8 +623,175 @@ def _submit_dataset_and_register_mappings(
+            x.dimension_type: x for x in project_config.model.dimensions.base_dimension_references
+        }
+        mapping_tables = []
+        for mapping in mappings:
+            mapping_table = run_in_other_dir(
+                src_dir,
+                MappingTableModel.from_pre_registered_model,
+                mapping,
+                dataset_mapping[mapping.dimension_type],
+                project_mapping[mapping.dimension_type],
+            )
+            mapping_tables.append(mapping_table)
+
+        mappings_config = DimensionMappingsConfig.load_from_model(
+            DimensionMappingsConfigModel(mappings=mapping_tables), src_dir
+        )
+        mapping_ids = self._dimension_mapping_mgr.register_from_config(
+            mappings_config, submitter, log_message, context=context
+        )
+        for mapping_id in mapping_ids:
+            mapping_config = self._dimension_mapping_mgr.get_by_id(mapping_id)
+            references.append(
+                DimensionMappingReferenceModel(
+                    from_dimension_type=mapping_config.model.from_dimension.dimension_type,
+                    to_dimension_type=mapping_config.model.to_dimension.dimension_type,
+                    mapping_id=mapping_id,
+                    version=str(self._dimension_mapping_mgr.get_current_version(mapping_id)),
+                )
+            )
+
+        return references
+
+    def _auto_register_reverse_supplemental_mappings(
+        self,
+        project_config: ProjectConfig,
+        dataset_config: DatasetConfig,
+        mapping_references: List[DimensionMappingReferenceModel],
+        autogen_reverse_supplemental_mappings: set[str],
+        submitter,
+        log_message,
+        context,
+    ):
+        references = []
+        p_model = project_config.model
+        p_base_dim_ids = {x.dimension_id for x in p_model.dimensions.base_dimension_references}
+        p_supp_dim_ids = {
+            x.dimension_id for x in p_model.dimensions.supplemental_dimension_references
+        }
+        d_dim_from_ids = set()
+        for ref in mapping_references:
+            mapping_config = self._dimension_mapping_mgr.get_by_id(ref.mapping_id)
+            d_dim_from_ids.add(mapping_config.model.from_dimension.dimension_id)
+
+        needs_mapping = []
+        for dim in dataset_config.model.dimension_references:
+            if dim.dimension_id not in p_base_dim_ids and dim.dimension_id not in d_dim_from_ids:
+                if (
+                    dim.dimension_id in p_supp_dim_ids
+                    and dim.dimension_type in autogen_reverse_supplemental_mappings
+                ):
+                    needs_mapping.append((dim.dimension_id, dim.version))
+                # else the dataset may only need to provide a subset of records, and those are
+                # checked in the dimension association table.
+
+        if len(needs_mapping) != len(autogen_reverse_supplemental_mappings):
```

Comment:
In cases where one project had successfully registered a reverse mapping, and another dataset could reuse that reverse mapping, in theory, user cannot use the --autogen flag anymore and need to instead reference that mapping, right?